\begin{center}
Gabe Morris
\end{center}

In [1]:
# Notebook Preamble
import matplotlib.pyplot as plt
import numpy as np

plt.style.use('maroon_ipynb.mplstyle')

\pagebreak
\tableofcontents
\pagebreak

\begin{center}
\begin{tabular}{c c c}
ME 8253 & Homework 6 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

#
## Given
A stepped circular rod of 4340 steel (with $S_u$ = 1468 MPa) with diameters of 60 and 45 mm has a root radius of 3 mm at the stepped section. The rod is to be subjected to axial cyclic loading.

![image not found](images/f1.PNG)

The cyclic yield strength ($S^\prime_y$) is estimated from:

$$S^\prime_y=K^\prime(0.002)^{n^\prime}$$

where $K^\prime$ and $n^\prime$ are given in Table A.2

For the purpose of constructing Haigh diagram, exact value of $\sigma_f$ is not needed, as the diagram is not very sensitive to its value. Since $\sigma_f$ is not listed in Table A.2 for this material, we use Eq. 5.20 in the textbook to approximate it as

$$\sigma_f\approx S_u + 345\ (MPa)$$

## Find
Using a Haigh diagram, determine the following for an approximate median fatigue life of $10^6$ cycles:

a. What fully reversed alternating force, $P_a$, can be applied?
b. What is the maximum value of $P_a$, if proper compressive residual stresses are present at the notch root? What is the magnitude of the compressive residual stress needed to obtain this maximum alternating stress?
c. What value of $P_a$ can be applied if the residual stress calculated in (b) is tensile? What fully reversed alternating force, $P_a$, can be applied?

## Solution
According to equation 4.3b, the endurance limit is $700\ MPa$ for materials with an ultimate strength greater than $1400\ MPa$. With the size effect, the endurance limit becomes

$$S_f=0.85(700)=595\ MPa$$

### Part A

![image not found](images/f2.png)

In [2]:
D, d, r = 60, 45, 3  # mm
Su = 1468  # MPa
size_effect = 0.85

if Su <= 1400:
    Sf = 0.5*Su*size_effect
else:
    Sf = 700*size_effect

D/d

1.3333333333333333

In [3]:
r/d

0.06666666666666667

From above the stress concentration factor is $K_t=2$. The fatigue notch factor for the fully reversed condition is

$$K_f=1+\frac{K_t-1}{1+a/r}$$

where $a=0.0254\left(\frac{2070}{S_u}\right)^{1.8}$ with $a$ in mm and $S_u$ in MPa.

In [4]:
Kt = 2
a = 0.0254*(2070/Su)**1.8
a  # mm

0.047149103389883054

In [5]:
Kf = 1 + (Kt - 1)/(1 + a/r)
Kf

1.9845268144780213

Since the stress is maximized at the smaller diameter, 

$$P_a=\frac{S_f}{K_f}\left(\frac{\pi}{4}\right)(d)^2$$

In [6]:
Pa = Sf/Kf*np.pi/4*d**2
Pa  # N

476842.4418454644

### Part B
From Table A.2,

| Property     | Value     |
|--------------|-----------|
| $S_y$        | 1371  MPa |
| $S^\prime_y$ | 863  MPa  |
| $S_f$        | 595   MPa |
| $\sigma_f$   | 1813  MPa |
| $K_f$        | 1.98      |
| $S_{cat}$    | 70    MPa |

$S_{cat}$ comes from the fact that this is a hard steel. The other calculations are shown below.

In [7]:
sig_f = Su + 345
sig_f  # MPa

1813

In [8]:
K_prime, n_prime = 1996, 0.135
Sy_prime = K_prime*0.002**n_prime
Sy_prime  # MPa

862.5804014077875

![image not found](images/f3.png)

In [9]:
Sa = 460  # MPa
Pa = Sa*np.pi/4*d**2
Pa  # N

731598.3892047232

### Part C
From above, point B is when the mean stress is tensile.

In [10]:
Sa = 90
Pa = Sa*np.pi/4*d**2
Pa  # N

143138.81527918496

\pagebreak

\begin{center}
\begin{tabular}{ccc}
ME 8253 & Homework 6 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

#
## Given
An axially loaded member made of 2024-T3 aluminum is repeatedly subjected to the block of stress history shown below.

![image not found](images/f4.PNG)

## Find
a. Complete a summary of the loading block.
b. Using the Basquin equation, $S_{Nf}=\sigma^\prime_f(2N_f)^b$, determine the fatigue strength $S_{Nf}$, the fatigue life $N_f$ and the damage ratio $n/N_f$ for each load segment and estimate the expected life if the member is smooth.
c. Estimate the expected life if the member has a notch with $K_t=2$ and the notch root radius is 1 mm. For the notched member, assume that the given nominal stress block and $K_t$ are based on net stress.

## Solution
### Part A
| Load Segment | $S_{min}$ (MPa) | $S_{max}$ (MPa) | $S_a$ (MPa) | $S_m$ (MPa) | n   |
|--------------|-----------------|-----------------|-------------|-------------|-----|
| 1            | 0               | 300             | 150         | 150         | 100 |
| 2            | -300            | 300             | 300         | 0           | 1   |
| 3            | -300            | 0               | 150         | -150        | 100 |

### Part B
Use the modified Goodman to find $S_{Nf}$ with $S_u=469$ MPa.

$$\frac{S_a}{S_{Nf}}+\frac{S_m}{S_u}=1\rightarrow S_{Nf}=\frac{S_aS_u}{S_u-S_m}$$
$$S_{Nf}=\sigma^\prime_f(2N_f)^b\rightarrow N_f=\frac{1}{2}\left(\frac{S_{Nf}}{\sigma^\prime_f}\right)^{1/b}$$

In [11]:
Su, sig_prime, b = 469, 1100, -0.124

SNf_lamb = lambda Sa_, Su_, Sm_: Sa_*Su_/(Su_ - Sm_)
Nf_lamb = lambda SNf_, sig_prime_, b_: 0.5*(SNf_/sig_prime_)**(1/b_)

# Load 1
SNf1 = SNf_lamb(150, Su, 150)
SNf1  # MPa

220.53291536050156

In [12]:
Nf1 = Nf_lamb(SNf1, sig_prime, b)
Nf1

212496.20843121517

In [13]:
d1 = 100/Nf1
d1

0.0004705966319976477

In [14]:
# Load 2
SNf2 = SNf_lamb(300, Su, 0)
SNf2  # MPa

300.0

In [15]:
Nf2 = Nf_lamb(SNf2, sig_prime, b)
Nf2

17764.216450750755

In [16]:
d2 = 1/Nf2
d2

5.629294164324021e-05

In [17]:
# Load 3
SNf3 = SNf_lamb(150, Su, -150)
SNf3  # MPa

113.65105008077545

In [18]:
Nf3 = Nf_lamb(SNf3, sig_prime, b)
Nf3

44578464.41972726

In [19]:
d3 = 100/Nf3
d3

2.24323563634792e-06

Now the expected life is calculated as the reciprocal of the summation of the damage ratios.

In [20]:
1/sum([d1, d2, d3])  # Blocks

1889.8846990152454

| Load Segment | $S_{Nf}$ (MPa) | $N_f$      | $n$ | $n/N_f$             |
|--------------|----------------|------------|-----|---------------------|
| 1            | 221            | 212496     | 100 | $4.706\cdot10^{-4}$ |
| 2            | 300            | 17764      | 1   | $5.630\cdot10^{-5}$ |
| 3            | 114            | 44,578,464 | 100 | $2.243\cdot10^{-6}$ |
| Total        | -              | -          | -   | 1890                |

### Part C
Everything remains the same, but now the endurance limit is changed to $S_f/K_f$, and a new $b$ is found by assuming that the endurance limit is reached at $10^6$ cycles.

In [21]:
a = 0.0254*(2070/Su)**1.8
a  # mm

0.3676793350247542

In [22]:
Kt, r = 2, 1
Kf = 1 + (Kt - 1)/(1 + a/r)
Kf

1.7311655403361788

In [23]:
Sf = sig_prime*2e6**b
Sf  # MPa

181.9973086280446

In [24]:
Sf_new = Sf/Kf
B = np.log(Sf_new/sig_prime)/np.log(2e6)
B

-0.16182533948270703

The $S_{Nf}$ values stay the same, but the $N_f$ values change.

In [25]:
Nf1 = Nf_lamb(SNf1, sig_prime, B)
Nf1

10274.56557852558

In [26]:
d1 = 100/Nf1
d1

0.009732771593672595

In [27]:
Nf2 = Nf_lamb(SNf2, sig_prime, B)
Nf2

1534.2156409133563

In [28]:
d2 = 1/Nf2
d2

0.0006517988562577001

In [29]:
Nf3 = Nf_lamb(SNf3, sig_prime, B)
Nf3

617791.4135432595

In [30]:
d3 = 100/Nf3
d3

0.00016186693082453747

In [31]:
1/sum([d1, d2, d3])  # Blocks

94.81874910904062

| Load Segment | $S_{Nf}$ (MPa) | $N_f$  | $n$ | $n/N_f$             |
|--------------|----------------|--------|-----|---------------------|
| 1            | 221            | 10275  | 100 | $9.732\cdot10^{-3}$ |
| 2            | 300            | 1534   | 1   | $6.518\cdot10^{-4}$ |
| 3            | 114            | 617791 | 100 | $1.619\cdot10^{-4}$ |
| Total        | -              | -      | -   | 95                  |